src/index.js

```diff
@@ -248,6 +248,10 @@ function shimHasOne(target) {
 
       return Promise.resolve().then(() => {
         const sourceKey = instance.get(this.sourceKey);
+        
+        if(!sourceKey) {
```

Comment:
must be an explicit null check or the test for `0` fails.

Please also add a test for this case.

Reply:
@lewismclennan That test must not be covering all cases if this fix is needed? Please add a test that covers this fix so it's regression proof.